src/main/java/com/thebois/Pawntastic.java

```diff
@@ -162,15 +182,24 @@ private void generateFont() {
         generator.scaleForPixelHeight(DEFAULT_FONT_SIZE);
         font = generator.generateFont(parameter);
         // To smooth out the text.
-        font.getRegion().getTexture().setFilter(Texture.TextureFilter.Linear,
-                                                Texture.TextureFilter.Linear);
+        font.getRegion()
+            .getTexture()
+            .setFilter(Texture.TextureFilter.Linear, Texture.TextureFilter.Linear);
         generator.dispose();
     }
 
     private void loadModelsFromSaveFile() throws IOException, ClassNotFoundException {
         final LoadSystem loadSystem = new LoadSystem();
         world = loadSystem.loadWorld();
         colony = loadSystem.loadColony();
+
+        RoleFactory.setWorld(world);
+        RoleFactory.setResourceFinder(world);
+        RoleFactory.setStructureFinder(world);
+
+        final IPathFinder pathFinder = new AstarPathFinder(world);
+        ActionFactory.setPathFinder(pathFinder);
```

Comment:
Ah, I see.

Yeah, since the factories don't get saved to the save file. The factories are instantiated from scratch again when the program is closed and opened, and then their static fields are null again. So they need to be called to be re-set.

Reply:
We will see how that works out for stockpiles then 😬